static/slider.js

```diff
@@ -0,0 +1,148 @@
+ *         sublevelID: [Array with timestamps from `observation_date`]
+ *     }
+ *   }
+ * }
+ * 
+ * 2. When sliding change value current displayed date.
+ * 
+ * 3. When value in slider change, filter parsed `observation` using current
+ *   date from slider comparing two timestamps. In addition calculate new quantity of
+ *   filtered observations.
+ * 
+ * 4. Use filtered observations and new quantity of observations to calculate new color for cell.
+ * 
+ * 5. Change color in cells and value below them.
+ * 
+ */
+
+(() => {
+    const observations = JSON.parse(window.observations) || {};
+    const COLORS_DARK = JSON.parse(window.COLORS_DARK)
+
+    $('#date-slider').slider({
+        min: dateToTime(window.minDate),
+        max: dateToTime(window.maxDate),
+        change: change,
+        slide: slide
+    });
+
+    // Set start value to min date available in slider.
+    $('#slider-value').html(`Current date: ${formatDate($('#date-slider').slider("option", "min"))}`);
+
+    /**
+     * Changes element which displays current date. Called when moving slider.
+     */
+    function slide(event, ui) {
+        $("#slider-value").html(`Current date: ${(formatDate(ui.value))}`);
+    }
+
+    /**
+     * Function called when value in slider change.
+     * Filters observations by current date on slider.
+     */
+    function change(event, ui) {
+        const timestamp = dateToTime(ui.value);
+
+        var observationsFiltered = {};
+        var allStars = {};
+
+        for (var construct in observations) {
+            observationsFiltered[construct] = {};
+
+            if (!allStars.hasOwnProperty(construct)) {
+                allStars[construct] = 0;
+            }
+
+            for (var course in observations[construct]) {
+                observationsFiltered[construct][course] = {}
+
+                for (var sublevel in observations[construct][course]) {
+                    observationsFiltered[construct][course][sublevel] = observations[construct][course][sublevel]
+                        .filter(date => dateToTime(date) > timestamp);
+
+                    allStars[construct] += observationsFiltered[construct][course][sublevel].length;
+                }
+            }
+        }
+
+        changeElements(observationsFiltered, allStars);
+    }
+
+    /**
+     * Changes colored cells and those with observations count after filtering observations by date.
+     * @param {Object} observationsFiltered - Observations filtered by date.
+     * @param {Object} allStars - Observations count per construct.
+     */
+    function changeElements(observationsFiltered, allStars) {
+        for (var construct in observationsFiltered) {
+            for (var course in observationsFiltered[construct]) {
+                for (var sublevel in observationsFiltered[construct][course]) {
+                    const color = calculateNewColor4(
+                        observationsFiltered[construct][course][sublevel].length,
```

Comment:
This expression is quite long. Please assign it to variable. It will be some extra documentation aspect.